src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -15,9 +18,38 @@ public void WriteResource(CodeWriter writer, Resource resource)
             using (writer.Namespace(cs.Namespace))
             {
                 writer.WriteXmlDocumentationSummary(resource.Description);
+
                 using (writer.Scope($"{resource.Declaration.Accessibility} class {cs.Name} : {cs.Name}Operations"))
                 {
-                    // Write Data property
+                    // internal constructor
+                    writer.WriteXmlDocumentationSummary($"Initializes a new instance of the <see cref = \"{cs.Name}\"/> class.");
+                    writer.WriteXmlDocumentationParameter("options", "The client parameters to use in these operations.");
+                    writer.WriteXmlDocumentationParameter("resource", "The resource that is the target of operations.");
+                    using (writer.Scope($"internal {cs.Name}({typeof(ResourceOperationsBase)} options, {cs.Name}Data resource) : base(options, resource.Id)"))
```

Comment:
It would be better if we had a reference to the ResourceData instance (https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Output/ResourceData.cs#L13) from the `Resource` and used its name for printing instead of concatenating it here. That way we would plug into existing generator features like type renaming, etc.